application-mocca-calendar-api/src/main/java/org/xwiki/contrib/moccacalendar/internal/importJob/ImportJob.java

```diff
@@ -0,0 +1,266 @@
+            documentReferenceResolver.resolve(EventConstants.MOCCA_CALENDAR_EVENT_MODIFICATION_CLASS_NAME);
+
+        for (RecurrentEventModification eventModification : modList) {
+            BaseObject eventModObj = eventDoc.newXObject(eventRecModifiedRef, wikiContext);
+            eventModObj.set(EventConstants.PROPERTY_ORIG_STARTDATE_OF_MODIFIED_NAME,
+                eventModification.getOriginalDate(), wikiContext);
+            eventModObj.set(EventConstants.PROPERTY_STARTDATE_NAME, eventModification.getModifiedStartDate(),
+                wikiContext);
+            eventModObj.set(EventConstants.PROPERTY_ENDDATE_NAME, eventModification.getModifiedEndDate(), wikiContext);
+            addConvertedPropertyToObject(eventModification.getModifiedTitle(), eventModObj,
+                Syntax.XWIKI_2_1.toIdString(), EventConstants.PROPERTY_TITLE_NAME);
+
+            addConvertedPropertyToObject(eventModification.getModifiedDescription(), eventModObj,
+                Syntax.XWIKI_2_1.toIdString(), EventConstants.PROPERTY_DESCRIPTION_NAME);
+        }
+    }
+
+    private byte[] getICSFileContent(byte[] importedFileContent)
+    {
+        String importedFileString = new String(importedFileContent);
+        int beginIndex = importedFileString.indexOf(CalendarKeys.ICS_CALENDAR_CALENDAR_BEGIN);
+        int endIndex = importedFileString.indexOf(CalendarKeys.ICS_CALENDAR_CALENDAR_END)
+            + CalendarKeys.ICS_CALENDAR_CALENDAR_END.length();
+        String filteredCalendarContent = importedFileString.substring(beginIndex, endIndex);
+
+        return filteredCalendarContent.getBytes();
+    }
+
+    private XWikiDocument getUniqueEventName(String eventName, String parentRef, List<XWikiDocument> eventDocuments)
+        throws XWikiException
+    {
+        XWikiContext wikiContext = this.wikiContextProvider.get();
+
+        DocumentReference parentDocRef = documentReferenceResolver.resolve(parentRef);
+        SpaceReference documentSpaceRef = new SpaceReference(eventName, parentDocRef.getLastSpaceReference());
+        DocumentReference eventRef = new DocumentReference(NEW_EVENT_HOME, documentSpaceRef);
+        wikiContext.getWiki().getDocument(eventRef, wikiContext);
+        XWikiDocument eventDoc = wikiContext.getWiki().getDocument(eventRef, wikiContext);
+
+        if (!eventDoc.isNew() || sameNameEventExists(eventRef, eventDocuments)) {
+            String newEventName = eventName + "_" + System.nanoTime();
+            SpaceReference newDocumentSpaceRef = new SpaceReference(newEventName, parentDocRef.getLastSpaceReference());
+            DocumentReference newEventRef = new DocumentReference(NEW_EVENT_HOME, newDocumentSpaceRef);
+            eventDoc = new XWikiDocument(newEventRef);
+        }
+        eventDoc.setTitle(eventName);
+        eventDoc.setParentReference(parentDocRef);
+        return eventDoc;
+    }
+
+    private boolean sameNameEventExists(DocumentReference eventRef, List<XWikiDocument> eventDocuments)
```

Comment:
There is an `equals` method between 2 XWikiDocuments, maybe you can create a XWikiDocument from `eventRef` and simply use a `contains` method, to not iterate manually?

Reply:
I tried to use the `contains` method for a created `XWikiDocument` from the `eventRef`, but it did not work properly.